app/assets/scripts/components/NewIndicator.js

```diff
@@ -30,12 +40,20 @@ class NewIndicator extends React.Component {
     });
   }
 
+  handleChange (e) {
```

Comment:
Since there's duplication between NewIndicator and EditIndicator, do you think this textfield should be its own widget?